torba/torba/client/baseheader.py

```diff
@@ -99,6 +103,37 @@ def hash_header(header: bytes) -> bytes:
             return b'0' * 64
         return hexlify(double_sha256(header)[::-1])
 
+    @asynccontextmanager
+    async def checkpointed_connector(self):
+        buf = BytesIO()
+        buf.connect = lambda _, headers: buf.write(headers)
```

Comment:
Is this used anywhere? And if so, why can't whatever uses it call `write` instead? If it is necessary, it would be less hacky to make a class inheriting BytesIO